src/server/qtquick/private/wquickinputmethodv2.cpp

```diff
@@ -214,8 +212,8 @@ WQuickInputPopupSurfaceV2::WQuickInputPopupSurfaceV2(QWInputPopupSurfaceV2 *hand
 
 WSurface *WQuickInputPopupSurfaceV2::surface() const
 {
-    W_DC(WQuickInputPopupSurfaceV2);
-    return d->popupSurface;
+    WSurface *wSurface = WSurface::fromHandle(handle()->surface());
+    return wSurface ? wSurface : new WSurface(handle()->surface());
```

Comment:
The new WSurface should following  `handle()->surface()` to destroy.

Reply:
Done.

Reply:
Remains constructing WSurface in private data's constructor but first try to look up from existent ones.